tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +95,364 @@ mvn package
+- [Retrieve airlines by ID](#get-route).
+- [List all airlines with pagination](#get-route).
+- [List airlines by country](#custom-sql-queries).
+- [List airlines by destination airport](#custom-sql-queries).
 
-## Document Structure
+### Document Structure
 
-We will be setting up a REST API to manage some profile documents. Our profile document will have an auto-generated UUID for its key, first and last name of the user, an email, and hashed password. For this demo we will store all profile information in just one document in a collection named `profile`:
+We will be setting up a REST API to manage some airline documents. Our airline document will have a structure similar to the following:
 
 ```json
 {
-  "pid": "b181551f-071a-4539-96a5-8a3fe8717faf",
-  "firstName": "John",
-  "lastName": "Doe",
-  "email": "john.doe@couchbase.com",
-  "password": "$2a$10$tZ23pbQ1sCX4BknkDIN6NekNo1p/Xo.Vfsttm.USwWYbLAAspeWsC"
+  "id": "airline_8091",
+  "type": "airline",
+  "name": "Couchbase Airways",
+  "callsign": "Couchbase",
+  "iata": "CB",
+  "icao": "CBA",
+  "country": "United States",
+  "active": true
 }
 ```
 
-As we can see, we want our user's password to be encrypted in the database too, we can achieve this simply with bcrypt, a dependency we have installed.
+The `id` field is the unique identifier for the document. The `type` field is used to identify the type of document. The `name` field is the name of the airline. The `callsign` field is the callsign of the airline. The `iata` field is the IATA code of the airline. The `icao` field is the ICAO code of the airline. The `country` field is the country of the airline. The `active` field is a boolean value indicating whether the airline is active or not.
+
+### Code Organization
+
+- `src/test/java`: Contains integration tests.
+- `src/main/java/org/couchbase/quickstart/springboot/repositories`: Contains the repository implementation.
+- `src/main/java/org/couchbase/quickstart/springboot/models`: Contains the data model.
+- `src/main/java/org/couchbase/quickstart/springboot/controllers`: Contains the RESTful API controllers.
+- `src/main/java/org/couchbase/quickstart/springboot/services`: Contains the service classes.
+
+## Data Model
+
+For this tutorial, we use three collections, `airport`, `airline` and `route` that contain sample airports, airlines and airline routes respectively. The route collection connects the airports and airlines as seen in the figure below. We use these connections in the quickstart to generate airports that are directly connected and airlines connecting to a destination airport. Note that these are just examples to highlight how you can use SQL++ queries to join the collections.
+
+![Data Model](./travel_sample_data_model.png)
 
 ## Let's Review the Code
 
 To begin clone the repo and open it up in the IDE of your choice to learn about how to create, read, update and delete documents in your Couchbase Server.
 
-## POST a Profile
+### Integration Tests
```

Comment:
Please move the tests to the end. Also needs a bit of explanation on how to run the tests and what the tests are doing.

Reply:
moved the tests to the end. Also added explaination for them